sdk/language-support/ts/codegen/src/TsCodeGenMain.hs

```diff
@@ -404,15 +419,15 @@ renderTemplateNamespace TemplateNamespace{..} = T.unlines $ concat
     tI = "typeof " <> tnsName <.> "templateId"
     tParams xs = "<" <> T.intercalate ", " xs <> ">"
 
-data TemplateRegistration = TemplateRegistration T.Text
+data TemplateRegistration = TemplateRegistration T.Text PackageId PackageReference
 
 renderTemplateRegistration :: TemplateRegistration -> T.Text
-renderTemplateRegistration (TemplateRegistration t) = T.unlines
-  [ "damlTypes.registerTemplate(exports." <> t <> ");" ]
+renderTemplateRegistration (TemplateRegistration t pkgId pkgRef) = T.unlines
+  [ "damlTypes.registerTemplate(exports." <> t <> ", ['" <> unPackageId pkgId <> "', '" <> forTemplateId pkgRef <> "']);" ]
```

Comment:
The typescript function `registerTemplate` now has an extra optional arg, where you can list alternatives for the package descriptor. This ensures that the template is registered against template ids with both the package id and the package name, so that either can be interpreted.